provider/machinepool/hcp/machine_pool_resource.go

```diff
@@ -1150,6 +1157,13 @@ func populateState(ctx context.Context, object *cmv1.NodePool, state *HcpMachine
 		} else {
 			state.AWSNodePool.AdditionalSecurityGroupIds = types.ListNull(types.StringType)
 		}
+		if httpTokensState, ok := awsNodePool.GetEc2MetadataHttpTokens(); ok {
+			if httpTokensState != "" {
+				state.AWSNodePool.Ec2MetadataHttpTokens = types.StringValue(string(httpTokensState))
+			} else {
+				state.AWSNodePool.Ec2MetadataHttpTokens = types.StringValue(ec2.HttpTokensStateOptional)
+			}
```

Comment:
```suggestion
                        state.AWSNodePool.Ec2MetadataHttpTokens = types.StringValue(ec2.HttpTokensStateOptional)
			if httpTokensState != "" {
				state.AWSNodePool.Ec2MetadataHttpTokens = types.StringValue(string(httpTokensState))
			}
```
np: trying to avoid elses when possible

Reply:
@gdbranco 
thanks for the comment, please help to re-review the latest commit, thanks